features/facebookImport/src/model/importers/multiple-files-importer.js

```diff
@@ -0,0 +1,81 @@
+import { createErrorResult, IMPORT_SUCCESS } from "./utils/importer-status";
+import {
+    readFullPathJSONFile,
+    relevantZipEntries,
+    removeEntryPrefix,
+    sliceIntoChunks,
+} from "./utils/importer-util";
+
+export default class MultipleFilesImporter {
+    // eslint-disable-next-line no-unused-vars
+    _isTargetPostFile(entryName) {
+        return false;
+    }
+
+    // eslint-disable-next-line no-unused-vars
+    _importRawDataResults(facebookAccount, dataResults) {}
+
+    async _extractTargetEntries(zipFile) {
+        const entries = await relevantZipEntries(zipFile);
+        return entries.filter((fileName) => this._isTargetPostFile(fileName));
+    }
+
+    async _readJSONFileWithStatus(targetFile, zipFile) {
+        return readFullPathJSONFile(targetFile, zipFile)
+            .then((data) => {
+                return { status: IMPORT_SUCCESS, targetFile, data };
+            })
+            .catch((error) => {
+                return createErrorResult(this.constructor, error);
+            });
+    }
+
+    async _importDataFromFiles(targetFiles, zipFile, facebookAccount) {
+        const fileDataWithResults = await Promise.all(
+            targetFiles.map((targetFile) =>
+                this._readJSONFileWithStatus(targetFile, zipFile)
+            )
+        );
+
+        const successfullResults = fileDataWithResults.filter(
+            (result) => result.status === IMPORT_SUCCESS
+        );
+
+        for (const each of successfullResults) {
+            const fileName = removeEntryPrefix(each.targetFile);
+            facebookAccount.addImportedFileName(fileName);
+        }
+        const dataResults = successfullResults.map((result) => result.data);
+        this._importRawDataResults(facebookAccount, dataResults);
+
+        return fileDataWithResults.filter(
+            (result) => !(result.status === IMPORT_SUCCESS)
+        );
+    }
+
+    _createMissingFilesError() {
+        return new Error("Missing import files");
```

Comment:
So you have no specific Exception in the base class, but you have it on derived classes? Any reason why?

Reply:
But what the derived classes provide is not actually related to the derived class, it's just a missing files exception, is that correct?

Reply:
At the moment they return different exceptions: `MissingMessagesFilesException` or `MissingPostFilesException`. But indeed, as we anyway associate the error with the importer in the report we can just have a generic `MissingFilesException`